editor/src/components/canvas/canvas-strategies/flex-resize-basic-strategy.tsx

```diff
@@ -0,0 +1,190 @@
+      const edgePosition = interactionState.activeControl.edgePosition
+      if (interactionState.interactionData.drag != null) {
+        const drag = interactionState.interactionData.drag
+        const originalBounds = MetadataUtils.getFrameInCanvasCoords(
+          selectedElement,
+          sessionState.startingMetadata,
+        )
+
+        if (originalBounds == null) {
+          return emptyStrategyApplicationResult
+        }
+
+        const resizedBounds = resizeWidthHeight(originalBounds, drag, edgePosition)
+
+        const elementParentBounds =
+          MetadataUtils.findElementByElementPath(sessionState.startingMetadata, selectedElement)
+            ?.specialSizeMeasurements.immediateParentBounds ?? null
+
+        const resizeCommands = [
+          adjustCssLengthProperty(
+            'always',
+            selectedElement,
+            stylePropPathMappingFn('width', ['style']),
+            resizedBounds.width - originalBounds.width,
+            elementParentBounds?.width,
+            true,
+          ),
+          adjustCssLengthProperty(
+            'always',
+            selectedElement,
+            stylePropPathMappingFn('height', ['style']),
+            resizedBounds.height - originalBounds.height,
+            elementParentBounds?.height,
+            true,
+          ),
+        ]
+        return strategyApplicationResult([
+          ...resizeCommands,
+          updateHighlightedViews('mid-interaction', []),
+          setCursorCommand('mid-interaction', pickCursorFromEdgePosition(edgePosition)),
+          setElementsToRerenderCommand(selectedElements),
+        ])
+      } else {
+        return strategyApplicationResult([
+          updateHighlightedViews('mid-interaction', []),
+          setCursorCommand('mid-interaction', pickCursorFromEdgePosition(edgePosition)),
+        ])
+      }
+    }
+    // Fallback for when the checks above are not satisfied.
+    return emptyStrategyApplicationResult
+  },
+}
+export function resizeWidthHeight(
+  boundingBox: CanvasRectangle,
+  drag: CanvasPoint,
+  edgePosition: EdgePosition,
+): CanvasRectangle {
+  if (isEdgePositionACorner(edgePosition)) {
+    const startingCornerPosition = {
+      x: 1 - edgePosition.x,
+      y: 1 - edgePosition.y,
+    } as EdgePosition
+
+    let oppositeCorner = pickPointOnRect(boundingBox, startingCornerPosition)
+    const draggedCorner = pickPointOnRect(boundingBox, edgePosition)
+    const newCorner = offsetPoint(draggedCorner, drag)
+
+    const newWidth = Math.abs(oppositeCorner.x - newCorner.x)
+    const newHeight = Math.abs(oppositeCorner.y - newCorner.y)
+
+    return canvasRectangle({
+      x: boundingBox.x,
+      y: boundingBox.y,
+      width: newWidth,
+      height: newHeight,
+    })
+  } else {
+    const isEdgeHorizontalSide = isEdgePositionAHorizontalEdge(edgePosition)
+
+    const oppositeSideCenterPosition = isEdgeHorizontalSide
+      ? ({
+          x: edgePosition.x,
+          y: 1 - edgePosition.y,
```

Comment:
There's some utility functions like `oppositeEdgePositionPart`, which I _believe_ means you can avoid those casts.

Reply:
Good idea, fixed in f676396b2bef16107049579f9f56fbec7938aed5